src/modules/status.rs

```diff
@@ -109,10 +109,7 @@ fn format_exit_code<'a>(
         }
     };
 
-    let hex_status = exit_code
-        .parse::<i32>()
-        .ok()
-        .map(|code| format!("0x{:X}", code));
+    let hex_status = format!("0x{:X}", exit_code_int as u32);
```

Comment:
I think this will just work and avoid truncation:

```suggestion
    let hex_status = format!("0x{:X}", exit_code_int);
```

Reply:
Avoiding truncation makes hex_status longer when status code is negative number like this:
```
✖-1 0xFFFFFFFFFFFFFFFF ❯
```
On Windows, status code is in range of u32 or i32 (e.g, MSVC `exit()` receive `int`), so I think this cast is necessary.

Reply:
In that case, should this also apply to `int` too? Furthermore, maybe this should be restricted to Windows (with another `type ExitCodeTarget`?) and I feel like this should be documented in the `status` docs.

Reply:
Whether i32 or u32 the status fits in is consistent in one shell, and in supported platform the status code fits in 32bit.
Until the platform that have larger status code than 32bit will be supported, I think that assuming that the status code fits in 32bit is no problem and hex_status doesn't have to be restricted only for Windows. Similally, I think it doesn't have to cast like hex_status.
But I agree that the size of status code that the module receives should be documented.

Reply:
Instead of that, it is possible to make ExitCode i32.
```diff
diff --git a/src/modules/status.rs b/src/modules/status.rs
index 5f643a95..31d26c2c 100644
--- a/src/modules/status.rs
+++ b/src/modules/status.rs
@@ -6,7 +6,7 @@ use crate::configs::status::StatusConfig;
 use crate::formatter::{string_formatter::StringFormatterError, StringFormatter};
 use crate::segment::Segment;
 
-type ExitCode = i64;
+type ExitCode = i32;
 type SignalNumber = u32;
 #[derive(PartialEq)]
 enum PipeStatusStatus<'a> {
@@ -101,15 +101,16 @@ fn format_exit_code<'a>(
     config: &'a StatusConfig,
     context: &'a Context,
 ) -> Result<Vec<Segment>, StringFormatterError> {
-    let exit_code_int: ExitCode = match exit_code.parse() {
-        Ok(i) => i,
+    // First, parse as i64 to accept both i32 or u32, then normalize to u32.
+    let exit_code_int: ExitCode = match exit_code.parse::<i64>() {
+        Ok(i) => i as ExitCode,
         Err(_) => {
             log::warn!("Error parsing exit_code string to int");
             return Ok(Vec::new());
         }
     };
 
-    let hex_status = format!("0x{:X}", exit_code_int as u32);
+    let hex_status = format!("0x{:X}", exit_code_int);
 
     let common_meaning = status_common_meaning(exit_code_int);
 

```

Reply:
In that case, I think it's better to leave `int` as-is for now.

Reply:
Done, I pushed the diff to this PR. Thanks for reviewing!

Reply:
Please also add a note in the `status` module description in `docs/config/README.md` about this and that `hex_status` is unsigned.

Reply:
Done.